### DISCUSSION TITLE
How to involve more volunteers?

### DISCUSSION THREAD

**Discussion by todordev** [joomla/joomla-cms#31865] [General]
Everyone knows we need more volunteers to move forward faster. Let we share our ideas about how to do it.
I am going to share with you one of my own right now.

The idea is about a platform where everyone will be able to find goals for achievement. It should be installed on the volunteers portal and managed by [Volunteer Engagement Team](https://volunteers.joomla.org/teams/volunteer-engagement-team) (VET).
It should provide a form where everyone will be able to send proposal for a goal. The VET will review and categorize received proposal. They are also be able able to add additional information about the goal.

The result should look like a list with clearly described things that the community needs. Every item should provide information such detailed description, the name of the people that have been worked on it, status  (pending, completed), external links, screenshots, reactions (happy, thank you, awesome).

The Volunteer Engagement Team should select goals from the list and should create a newsletter. They should send the newsletter to the subscribers every two weeks. The newsletter should be reworked every time.

The main purpose is to provide a clear list of goals to everyone who would like to help. We should be able to introduce the needs of the project to the people. This systematic list will provide an opportunity for them to choose the ways in which they will contribute to the project.

You will have to send the first newsletter to everyone registered on JED, the community portal, the volunteers portal, the forum, the community magazine, the documentation portal, all mailing lists (Developer Network Newsletter, General Extensions Mailing, CMS Mailing, Framework Mailing, Documentation Mailing). You should publish an article on the magazine. You should share information on the social networks (twitter, reddit, facebook).

You should introduce the new system to the people and ask them to subscribe to the newsletter. If they want to help, they will subscribe and receive news regularly. They will build habits that will make them look forward to the list of new goals.

Furthermore, the VET can also contact the CIO of the largest companies that use Joomla! as base of their business. They should introduce them the new system and should ask them for help. If they would like to help, they could subscribe to the newsletter. They also will receive a place in the list with completed goals. It will be good advertise for them.

The newsletter should look like this one.
- - -
# Goals List
(01.01 - 08.01.2021)

## Pull Requests
(we need your help to test and merge the following pull requests)

- [4.0 RFC Remove 3 column layout and introduce fieldWidth attribute as layout hint](https://github.com/joomla/joomla-cms/pull/31810)
- [4.0 Fix Composer Deprecation Notices](https://github.com/joomla/joomla-cms/pull/31804)
- [Pre-Upgrade Compatibility Check on Plugins That Could Kill Joomla Upgrade](https://github.com/joomla/joomla-cms/pull/31200)

## Issues
(we need your help to find a solution for the following issues)

- [4.0 Media crop plugin](https://github.com/joomla/joomla-cms/issues/31773)
- [4.0 com_csp URL](https://github.com/joomla/joomla-cms/issues/31709)
- [4.0 Review Joomla Update Compatibility](https://github.com/joomla/joomla-cms/issues/31490)

## Documentation
(we need your help to provide documentation for the following pages)

- [Controllers](https://docs.joomla.org/Controllers)
- [Models](https://docs.joomla.org/Models)
- [Views](https://docs.joomla.org/Views)

## Translation
(we need your help to translate Joomla in the following languages)

- [Romania](https://crowdin.com/project/joomla-cms/ro)
- [Finnish](https://crowdin.com/project/joomla-cms/fi)
- [Polish](https://crowdin.com/project/joomla-cms/pl)

## Discussions
(please, share your ideas and suggestions in the following discussion)

- [Abstract graphics](https://github.com/joomla/joomla-cms/discussions/31815)
- [Better SEO options in edit articles](https://github.com/joomla/joomla-cms/discussions/15833)
- [Replace Bootstrap 4 with Bootstrap 5](https://github.com/joomla/joomla-cms/discussions/31765)

## Questions
(please, answer to the following questions in our forum and groups)

- [Joomla 4 - getActive Menu item in Controller](https://groups.google.com/g/joomla-dev-general/c/mN0uScH7_RQ)
- [What is best practice for inserting the link for custom fields in third pary extensions](https://groups.google.com/g/joomla-dev-general/c/MLI8ZFU8WSk)
- [Parameter for using custom fields](https://groups.google.com/g/joomla-dev-general/c/mL8IfQpZXWM)
- [Link options when redirecting](https://forum.joomla.org/viewtopic.php?f=706&t=983882)
- [Suggestion for creating a new free template for Joomla 4](https://forum.joomla.org/viewtopic.php?f=803&t=983867)

## Stack Overflow Questions
(please, give a hand to our community answering the following questions)

- [Managing authors of Joomla articles](https://joomla.stackexchange.com/questions/28774/managing-authors-of-joomla-articles)
- [Password Reset on Login Form](https://joomla.stackexchange.com/questions/28742/password-reset-on-login-form)
- [Link options when redirecting](https://joomla.stackexchange.com/questions/28773/link-options-when-redirecting)

Thank you very much in advance for your help!
- - -

We need to ask for help and we will get it. I am absolutely sure if we do that Joomla will get a lot of manpower. This year will become the year of Joomla. 

That was one of my ideas. What do you think?

**Comment by brianteeman** [comment]
excellent idea

**Comment by PhocaCz** [comment]
> That was one of my ideas. What do you think?

It's a great start, if we don't define what we need, no one will help us.

But, there is a 'but'. But along with this we also need to know what we can offer and how we can motivate others to help us.

It is not enough to tell large companies that they should return something. Most of them just get a CMS in both cases - even if they help, but even if they don't help. Most behave cynically (I don't mean it negatively) . So if there is no new version of Joomla! they will use current version of Joomla! or they switch to another CMS.

Yes, it's just my idea, but practice in recent years simply shows that people generally have a lot of own problems and in order to be a volunteer for a project, they need to be excited or motivated.

If Joomla 4 has some "Wow" feature, it's easier to get volunteers, but unfortunately Joomla 4 can be great inside, but we just lack the "Wow" feature on the "outside". Joomla! 4 does not include any significant feature which should attract new users/new volunteers. That is why we need to think about how we motivate others in another way.

The world is not idealistic, so mostly we ask people/firms who need help themselves (for example, if we talk about Joomla! 4, the fact is that they themselves have to completely rewrite everything for the new version). So there really aren't many volunteers "left in stock". 

Motivation is the key. If there was a boom with Joomla! 1.5, the motivation was just new modern built CMS with "Wow" effect. We are not in such a situation at the moment.

Motivate does not mean threatening in style: "if you don't help us, it can happen, then you will never use our CMS because it will be not finished". It will evoke the exact opposite. We really need to find out, what we can offer and how to inspire others to collaborate.

As I have written many times, this is nicely said, but even I myself just do not know how to achieve it, so I appreciate your ideas.

**Comment by leoalv** [comment]
One way to motivate people and companies is that all of us who are part of Joomla! (everyone from the simple user to the main developers) we concentrate on providing solutions to problems, they always focus on the problem (this will break, this changes, this delays, this is not necessary, this makes us look like wordpress, and a ton of buts).

@PhocaCz  You said it Joomla! 1.5 was WOW !, this was because they focused on troubleshooting and not rescuing the remains of Joomla 1.0.

The idea that @ITPrism is proposing is excellent and is focused on solving problems, if this becomes possible I think it will be the basis for solving many Joomla! Needs.

This is my opinion and my vision.

**Comment by richard67** [comment]
The idea of a newsletter to attract volunteers is a good idea, but it requires volunteers to make the newsletter .. a vicious circle.

**Comment by leoalv** [comment]
How to reach people? I guess everyone here knows or has seen the message to send confidential data for Joomla statistics (it comes out in every new installation). I propose a message the same but that is shown in the new installation and in each update of Joomla!. It should contain a very persuasive message with a link to the platform that @ITPrism is proposing where they can request a necessary feature in Joomla and invitation to join on the project as a VOLUNTEER. This request goes into voting to see if people really need it and obviously the review by the VET. With this we would work on what is really necessary and not on what a few of us think is necessary, and we would be complying with what Joomla means "All Together"

To this would be added the newsletters to JED registrants and direct contacts to the main Joomla template developers such as Joomlart, Joomshaper and others. I remember many years ago I saw a fundraising campaign for Joomlart, Rockettheme, shape5 (I don't know if they gave the money), but I think they would cooperate.

**Comment by PhocaCz** [comment]
>I propose a message the same but that is shown in the new installation and in each update of Joomla!. It should contain a very persuasive message with a link to the platform that @ITPrism is proposing where they can request a necessary feature in Joomla and invitation to join on the project as a VOLUNTEER

Just imagine, you are new to Joomla!. You install it for testing purposes, because you need a professional CMS. And first what you see, is a message asking you for volunteering. It's not very good to make a good impression.

>To this would be added the newsletters to JED registrants and direct contacts to the main Joomla template developers such as Joomlart, Joomshaper and others. I remember many years ago I saw a fundraising campaign for Joomlart, Rockettheme, shape5 (I don't know if they gave the money), but I think they would cooperate.

The idea that major Joomla! template developers have excess money and enough free time for volunteering, is even wrong. Just imagine, a template developer with 100+ templates who need to overwrite all of them for Joomla! 4 - these are incredibly high costs (a lot of people, a lot of time).  Many template developers reduced their staff, tried to earn money from other CMSs like WP or Grav, etc. :-(

**Reply by leoalv** [reply]
I don't know if I was clear, what I said is that a message is shown giving the option to request new features, and if he requests a feature already on the platform, he is invited to volunteer. If he installs it for testing, much better, he will see the message that tells him if you need something, request it, after being requested comes the invitation to volunteer, see if what he requests is necessary, it is possible and all the other filters.

From my point of view, if I install a CMS and he tells me if you need something, ask for it, I'll keep it.

What is being planned is not only for Joomla 4, I know it is the most urgent, but as you said everyone is busy with it, even me with my few modules and plugins that I have.
